tianshou/data/batch.py

```diff
@@ -859,6 +654,9 @@ def shape(self) -> List[int]:
                 try:
                     data_shape.append(v.shape)
                 except AttributeError:
+                    if is_scalar_value(v):
+                        data_shape.append([])
+                        continue
                     raise TypeError("No support for 'shape' method with "
```

Comment:
Is it really possible to reach this point ? Anyway, I would rather ALWAYS return `data_shape.append([])` in case of an exception, instead of raising an exception. Because I don't think `shape` should raise an exception under any condition:
`Batch(a=1.0).shape` must not fail, neither `Batch(a={1.0, 2.0}).shape` does.

Reply:
> I don't think shape should raise an exception under any condition:

That sounds reasonable. Even numpy scalars have shapes.

```python
>>> np.float64(1.0).shape
()
```

So yes, I agree with this change. What about @Trinkle23897 ?

Reply:
Agree

Reply:
Fixed in 6ec46c1 .